### ISSUE TITLE
Not able to do anything with file named "[test].md" (m, o , enter do not work)

### ISSUE THREAD

**Issue by netei** [scrooloose/nerdtree#492]
Whenever you name a file `[test].md`, it is not possible to do anything with it inside NerdTree (pressing m and any other keys has absolutely no effect)

This seems to come from the function UI._stripMarkup, specifically this line : https://github.com/scrooloose/nerdtree/blob/8c7534bc9e1d0c2f18bf94c1440c4ae6b102b5d6/lib/nerdtree/ui.vim#L359

```
function! s:UI._stripMarkup(line, removeLeadingSpaces)
    let line = a:line
    "remove the tree parts and the leading space
    let line = substitute (line, g:NERDTreeUI.MarkupReg(),"","")

    "strip off any read only flag
    let line = substitute (line, ' \[RO\]', "","")

    "strip off any bookmark flags
    let line = substitute (line, ' {[^}]*}', "","")

    "strip off any executable flags
    let line = substitute (line, '*\ze\($\| \)', "","")

    "strip off any generic flags
    let line = substitute (line, '\[[^]]*\]', "","")

    let wasdir = 0
    if line =~# '/$'
        let wasdir = 1
    endif
    let line = substitute (line,' -> .*',"","") " remove link to
    if wasdir ==# 1
        let line = substitute (line, '/\?$', '/', "")
    endif

    if a:removeLeadingSpaces
        let line = substitute (line, '^ *', '', '')
    endif

    return line
endfunction
```

I guess the changes that should be made to make this work would be huge (because the filename should not be extracted from the view layer, but from an other layer)

**Comment by lifecrisis** [comment]
This issue is directly related to #680. It is a confirmed bug. I independently discovered the problem with `_stripMarkup()` previously, and this user has reached the same conclusion.

**Comment by PhilRunninger** [comment]
@netei , would you be willing to test the pull request #868 I've put together for this issue? Details are in the PR. Thanks.

**Comment by PhilRunninger** [comment]
Fixed in #868.
